workers/data_refinery_workers/processors/management/commands/check_qn_reference_agreement.py

```diff
@@ -0,0 +1,117 @@
+# import csv
+# import json
+import os
+
+from django.conf import settings
+from django.core.management.base import BaseCommand
+
+import pandas as pd
+import requests
+import scipy.stats
+
+from data_refinery_common.logging import get_and_configure_logger
+from data_refinery_common.models import Organism
+from data_refinery_workers.processors import qn_reference
+from data_refinery_workers.processors.management.commands import create_qn_target
+
+logger = get_and_configure_logger(__name__)
+
+KS_TEST_THRESHOLD = 0.05
+
+
+def signal_failure(bad_organisms):
+    if not settings.RUNNING_IN_CLOUD or not settings.ENGAGEMENTBOT_WEBHOOK:
+        logger.info(
+            "Some newly-generated QN references were not similar-enough to our references currently in use.",
+            failing_organisms=", ".join(o.name for o in bad_organisms),
+        )
+        return
+
+    requests.post(
+        settings.ENGAGEMENTBOT_WEBHOOK,
+        json={
+            "attachments": [
+                {
+                    "fallback": "QN Reference Test Failed",
+                    "title": "QN Reference Test Failed",
+                    "color": "#db3b28",
+                    "text": "Some newly-generated QN references were not similar-enough to our references currently in use.",
+                    "fields": [
+                        {
+                            "title": "Failing Organisms",
+                            "value": ", ".join(o.name for o in bad_organisms),
+                        }
+                    ],
+                    "footer": "Refine.bio",
+                    "footer_icon": "https://s3.amazonaws.com/refinebio-email/logo-2x.png",
+                }
+            ],
+        },
+        headers={"Content-Type": "application/json"},
+        timeout=10,
+    )
+
+
+def check_qn_reference_agreement():
+    bad_organisms = []
+    for organism in Organism.objects.filter(qn_target__isnull=False):
+        final_context = create_qn_target.create_qn_target(
+            organism,
+            platform=create_qn_target.get_biggest_platform(organism),
+            create_results=False,
```

Comment:
`create_results` will only prevent the database objects from being created right? It doesn't prevent the files from being written to disk? Will these accumulate?

Reply:
Nice catch! I originally was using the files in the cron job test. Because I use the raw dataframe in the cron job test now, I added a check for `create_results` before writing files to the disk.